specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Trigger.json

```diff
@@ -422,6 +429,112 @@
           "maximum": 86400
         }
       }
+    },
+	  "TriggerReference": {
+      "description": "Trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Trigger reference type.",
+          "type": "string",
+          "enum": [
+            "TriggerReference"
+          ]
+        },
+        "referenceName": {
+          "description": "Reference trigger name.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type",
+        "referenceName"
+      ]
+    },
+    "DependencyReference": {
+      "description": "Referenced dependency.",
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "description": "The type of dependency reference.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "TriggerDependencyReference": {
+      "description": "Trigger referenced dependency.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DependencyReference"
+        }
+      ],
+      "properties": {
+        "referenceTrigger": {
+          "description": "Referenced trigger.",
+          "$ref": "#/definitions/TriggerReference"
+        }
+      },
+      "required": [
+        "referenceTrigger"
+      ]
+    },
+    "TumblingWindowTriggerDependencyReference": {
+      "description": "Referenced tumbling window trigger dependency.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TriggerDependencyReference"
+        }
+      ],
+      "properties": {
+        "offset": {
+          "description": "Timespan applied to the start time of a tumbling window when evaluating dependency, .Net timespan format.",
```

Comment:
Terms like ".Net timespan" will be very confusing to uses of clients generated in other languages. Please use official formatting terminology (e.g. ISO 8601).
If your API does not follow an official standard (which it should) then please supply an example of the formatting that needs to be supplied.

Is there a reason you're not using the swagger supported duration type rather than string?

Reply:
This property "offset" is input through UI. Users have option for offset in hours, and enter the number of hour users like. Internally we use .Net for this project, and parse the offset from UI into .net timespan.

Reply:
Thanks @haoyingli - but the swagger must be agnostic of any particular language - please remove references to .Net